### PR TITLE
chore: adding NIMFLAGS usage to readme

### PR DESCRIPTION
**Pull request by gabrielmer**
# Description
Added example in `README.md`of using custom nim compilation flags

# Changes

- [x] added example to `README.md`

## Issue
#2466

### CODE REVIEW

**Review by jakubgs — APPROVED** [review]
Nice. Maybe also a comment at first definition of `NIM_PARAMS` in the `Makefile`?

File: README.md [hunk 1]
```diff
@@ -26,6 +26,9 @@ The standard developer tools, including a C compiler, GNU Make, Bash, and Git. M
 # You'll run `make update` after each `git pull` in the future to keep those submodules updated.
 make wakunode2
 
+# Build with custom compilation flags
```
[the remark this answers is not in the document]
Comment (gabrielmer) [reply README.md]: Done :)

**Comment by gabrielmer** [comment]
> LGTM. Maybe if we would have a link to the possible flags?

That's a tough one, isn't there a ton of flags for nim + for every dependency that we use? Not sure where to take it from

**Comment by NagyZoltanPeter** [comment]
> > LGTM. Maybe if we would have a link to the possible flags?
> 
> That's a tough one, isn't there a ton of flags for nim + for every dependency that we use? Not sure where to take it from

You're right, that was more an innocent question. But wonder if this nimflag stuff mentioned here would really help one who first meet with waku, as in most of the cases we also don't use it for local Linux builds?

**Comment by gabrielmer** [comment]
> You're right, that was more an innocent question. But wonder if this nimflag stuff mentioned here would really help one who first meet with waku, as in most of the cases we also don't use it for local Linux builds?

I agree 100% that this isn't relevant for most users, even less for first time users. However, I think it's important to have documented how to set flags in case someone wants to.

The question is where is the right place to document it. Maybe in the repo's readme it might add noise to first time users (hopefully the comment helps avoid that). But we also don't want to have it hidden either.

So given the tradeoffs, not sure where should we have it. Any recommendations?

**Pull request merged**